### ISSUE TITLE
Several call warnings and "csound lib not found" when running "working" Csound examples

### ISSUE THREAD

**Issue by csounder** [BelaPlatform/Bela#612]
even when running the first "working" or any of the "working" Csound examples

you might want to note the warning messages?  are the OK?
- and
it probably is not OK that the program can't find the Csound plugins directory
- possibly a new name?

Building project ...
Build finished
Running project ...
   0"000.145| WARNING: [main] duplicate call from main program to xenomai_init() ignored
   0"000.382| WARNING: [main] (xeno-config --no-auto-init disables implicit call)
   0"063.997| WARNING: [main] duplicate call from main program to xenomai_init() ignored
   0"064.147| WARNING: [main] (xeno-config --no-auto-init disables implicit call)
WARNING: [mError opening plugin directory '/usr/local/lib/csound/plugins-6.0': No such file or directory[m
[m0dBFS level = 32768.0
--Csound version 6.14 (float samples) Feb  9 2020
[commit: 1073b4d1bc2304a1e06defd266781a9c441a5be0]
libsndfile-1.0.27
realtime mode enabled
UnifiedCSD:  /root/Bela/projects/exampleTempProject/_main.csd
STARTING FILE
Creating options
Creating orchestra
closing tag
Creating score
realtime mode enabled
Elapsed time at end of orchestra compile: real: 0.019s, CPU: 0.013s
sorting score ...
	... done
Elapsed time at end of score sort: real: 0.023s, CPU: 0.014s
setting dummy interface
graphics suppressed, ascii substituted
0dBFS level = 32768.0
orch now loaded
audio buffered in 256 sample-frame blocks
SECTION 1:
Initialising spinlock...
Starting realtime mode queue: 0xb4a21010 thread: 0x1f5a5c0
new alloc for instr 123:
Hello World!
backtrace() returned 0 addresses
[mCsound tidy up: Terminated
Makefile:581: recipe for target 'runide' failed
make: *** [runide] Terminated
inactive allocs returned to freespace
end of score.		   overall amps:   8192.0
	   overall samples out of range:        0
0 errors in performance
Elapsed time at end of performance: real: 31.963s, CPU: 1.150s
Bela stopped

**Comment by giuliomoro** [comment]
>  0"000.145| WARNING: [main] duplicate call from main program to xenomai_init() ignored
0"000.382| WARNING: [main] (xeno-config --no-auto-init disables implicit call)
0"063.997| WARNING: [main] duplicate call from main program to xenomai_init() ignored
0"064.147| WARNING: [main] (xeno-config --no-auto-init disables implicit call)

I *think* this is due to the way it's built. As suggested by the warning message, [this line](https://github.com/csound/csound/blob/develop/Bela/build.sh#L4) should have the `--no-auto-init` flag. I will test that.

> WARNING: �[mError opening plugin directory '/usr/local/lib/csound/plugins-6.0': No such file or directory�[m

This, I guess, is because we do not create that folder. What goes there in a normal csound installation? "Externals" that are shipped separately from the main `csound` executable? Would there be any advantage in creating that folder and leaving it empty?

**Comment by giuliomoro** [comment]
@csounder would you be able to answer my question above:

> What goes there in a normal csound installation? "Externals" that are shipped separately from the main csound executable? Would there be any advantage in creating that folder and leaving it empty?

**Comment by giuliomoro** [comment]
from #639 , Victor said:

> Yes, that could be the issue. Csound is probably installed somewhere else; that is the default plugin directory. The environment variables OPCODE6DIR64 or OPCODE6DIR can be used to fix this."

I checked the `.deb` packages that we use for Csound, and both the one I built (6.14) and the one Victor built previously (6.12) do not have that folder, however they both have the folder `/usr/lib/csound/plugins-6.0/`, which contains these files:
```
/usr/lib/csound/plugins-6.0/libampmidid.so
/usr/lib/csound/plugins-6.0/libarrayops.so
/usr/lib/csound/plugins-6.0/libbeosc.so
/usr/lib/csound/plugins-6.0/libbuchla.so
/usr/lib/csound/plugins-6.0/libcellular.so
/usr/lib/csound/plugins-6.0/libcontrol.so
/usr/lib/csound/plugins-6.0/libcs_date.so
/usr/lib/csound/plugins-6.0/libdoppler.so
/usr/lib/csound/plugins-6.0/libdssi4cs.so
/usr/lib/csound/plugins-6.0/libemugens.so
/usr/lib/csound/plugins-6.0/libexciter.so
/usr/lib/csound/plugins-6.0/libfareygen.so
/usr/lib/csound/plugins-6.0/libfractalnoise.so
/usr/lib/csound/plugins-6.0/libframebuffer.so
/usr/lib/csound/plugins-6.0/libftsamplebank.so
/usr/lib/csound/plugins-6.0/libgetftargs.so
/usr/lib/csound/plugins-6.0/libgtf.so
/usr/lib/csound/plugins-6.0/libimage.so
/usr/lib/csound/plugins-6.0/libipmidi.so
/usr/lib/csound/plugins-6.0/libjoystick.so
/usr/lib/csound/plugins-6.0/libliveconv.so
/usr/lib/csound/plugins-6.0/libmixer.so
/usr/lib/csound/plugins-6.0/libpadsynth.so
/usr/lib/csound/plugins-6.0/libplaterev.so
/usr/lib/csound/plugins-6.0/libpvsops.so
/usr/lib/csound/plugins-6.0/libpy.so
/usr/lib/csound/plugins-6.0/libquadbezier.so
/usr/lib/csound/plugins-6.0/librtalsa.so
/usr/lib/csound/plugins-6.0/librtpulse.so
/usr/lib/csound/plugins-6.0/libscansyn.so
/usr/lib/csound/plugins-6.0/libscugens.so
/usr/lib/csound/plugins-6.0/libselect.so
/usr/lib/csound/plugins-6.0/libserial.so
/usr/lib/csound/plugins-6.0/libsignalflowgraph.so
/usr/lib/csound/plugins-6.0/libstackops.so
/usr/lib/csound/plugins-6.0/libstdutil.so
/usr/lib/csound/plugins-6.0/libsystem_call.so
/usr/lib/csound/plugins-6.0/liburandom.so
```
does any of these opcodes work? If it does, then probably the folder is successfully found and searched for. Are any plugins missing that you could tell?

**Comment by giuliomoro** [comment]
As mentioned in https://github.com/BelaPlatform/Bela/issues/635, it would seem that the path is not actually searched. Try running this to fix it:

```
ln -s /usr/lib/csound/ /usr/local/lib/csound
```

**Comment by csounder** [comment]
YES!  This is the solution!!! A lot more of Csound now runs on the Bela.

*Dr. Richard Boulanger*

*Professor of Electronic Production and Design*

Professional Writing and Technology Division

*Berklee College of Music*


skype: radiobaton

facebook: https://www.facebook.com/richard.boulanger.58

about: http://www.boulangerlabs.com/#about

about: http://www.csounds.com/community/developers/dr-richard-boulanger/

music:
http://www.csounds.com/community/developers/dr-richard-boulanger/dr-richard-boulanger-music/


Boulanger Labs - http://boulangerlabs.com

The Csound Book - http://mitpress.mit.edu/books/csound-book

The Audio Programming Book -
http://mitpress.mit.edu/books/audio-programming-book
